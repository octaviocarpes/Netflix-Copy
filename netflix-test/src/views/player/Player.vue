<template>
  <div class="player-container">
    <div class="back-to-movies" @click="goToMovies()">
      <img :src="require(`../../assets/img/back-arrow.svg`)" alt="back icon">
      <p>Back to Movies</p>
    </div>
    <iframe
      :src="movieURL"
      frameborder="0"
      width="100%"
      height="100%"
      allowfullscreen
    ></iframe>
  </div>
</template>

<script>
export default {
  name: "Player",

  computed: {
    movieURL() {
      return `https://www.youtube.com/embed/${this.$route.params.id}?autoplay=1`
    }
  },

  methods: {
    goToMovies() {
      this.$router.push("/movies");
    }
  }
};
</script>

<style lang="scss" scoped>
.player-container {
  height: 100%;
  width: 100%;

  .back-to-movies {
    
    display: flex;

    position: absolute;
    z-index: 999;

    img {
      width: 100px;
      height: 100px;

      padding-top: 35px;
    }

    p {
      padding-top: 76px;
      color: white;

      &:hover {
        text-decoration: underline;
      }
    }

    &:hover {
      cursor: pointer;
    }
  }
}
</style>
