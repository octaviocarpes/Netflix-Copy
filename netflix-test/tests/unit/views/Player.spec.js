import Vue from "vue";
import Player from "@/views/player/Player.vue";

describe("Player.spec.js", () => {
  let cmp, vm;

  beforeEach(() => {
    cmp = Vue.extend(Player);
    vm = new cmp({}).$mount();
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});