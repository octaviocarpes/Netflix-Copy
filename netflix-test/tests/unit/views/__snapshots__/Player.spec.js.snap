// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player.spec.js has the expected html structure 1`] = `<!---->`;
